Server/ConcordiaCurriculumManager/Models/Curriculum/Course.cs

```diff
@@ -39,11 +39,36 @@ public class Course : BaseModel
 
     public CourseModificationRequest? CourseModificationRequest { get; set; }
 
+    public CourseDeletionRequest? CourseDeletionRequest { get; set; }
+
     // Self-reference related fields
     public ICollection<CourseReference>? CourseReferenced { get; set; }
 
     public ICollection<CourseReference>? CourseReferencing { get; set; }
-}
+
+    public static Course CloneCourseForDeletionRequest(Course course)
+    {
+        return new Course
+        {
+            Id = Guid.NewGuid(),
+            CourseID = course.CourseID,
+            Subject = course.Subject,
+            Catalog = course.Catalog,
+            Title = course.Title,
+            Description = course.Description,
+            CreditValue = course.CreditValue,
+            PreReqs = course.PreReqs,
+            Career = course.Career,
+            EquivalentCourses = course.EquivalentCourses,
+            CourseNotes = course.CourseNotes,
+            CourseState = CourseStateEnum.CourseDeletionProposal,
+            Version = course.Version + 1,
+            Published = course.Published,
+            CourseCourseComponents = course.CourseCourseComponents,
+            SupportingFiles = course.SupportingFiles,
```

Comment:
Worth noting that neither CourseCourseComponents nor SupportingFiles are correctly cloned, as you'll still be referencing the same rows in the DB, rather than creating new rows (it's a shallow copy instead of a deep copy). For our purposes here though, I don't think it'll matter.